### ISSUE TITLE
CSI: RPCs use ExternalID when set to address volumes

### ISSUE THREAD

**Issue by langmartin** [hashicorp/nomad#7302]
Our RPC layers are using just `ID` to address the volume when making controller & node plugin calls, they should use `ExternalID` in order to support ids that are not url safe (since we address volumes by `ID` in the api)

**Comment by tgross** [comment]
Probably related to this is that it seems like we can't have the ID and External ID separate -- if you try the registration below it works, but if we want to set the ID to something human readable for the volume instead of the UUID from the provider, it fails.

```sh
UUID=$(sudo csc --endpoint $CSI_ENDPOINT controller create-volume test-volume --cap 1,2,ext4 | grep -o '".*"' | tr -d '"')

echo "registering volume $UUID..."

echo $(printf 'id = "%s"
name = "test-volume"
type = "csi"
external_id = "%s"
plugin_id = "%s"
access_mode = "single-node-writer"
attachment_mode = "file-system"' $UUID $UUID $PLUGIN_ID) | nomad volume register -
```

**Comment by tgross** [comment]
Closed by #7326
